src/RealtimeEffectPanel.cpp

```diff
@@ -1109,8 +733,88 @@ class RealtimeEffectListWindow : public wxScrolledWindow
          }
       });
       SetScrollRate(0, 20);
+      
+      mPluginsChangedSubscription = PluginManager::Get().Subscribe(
+         [this](PluginsChangedMessage)
+         {
+            UpdateEffectMenuItems();
+         });
+      UpdateEffectMenuItems();
    }
 
+   std::optional<wxString> PickEffect(wxWindow* parent, const wxString& selectedEffectID)
+   {
+      wxMenu menu;
+      if(!selectedEffectID.empty())
+      {
+         //no need to handle language change since menu creates its own event loop
+         menu.Append(wxID_REMOVE, _("No Effect"));
+         menu.AppendSeparator();
+      }
+      
+      RealtimeEffectsMenuVisitor visitor { menu };
+      
+      Registry::Visit(visitor, mEffectMenuRoot.get());
+      
+      int commandId = wxID_NONE;
+      
+      menu.AppendSeparator();
+      menu.Append(wxID_MORE, _("Get more effects..."));
+      
+      menu.Bind(wxEVT_MENU, [&](wxCommandEvent evt) { commandId = evt.GetId(); });
+      
+      if(parent->PopupMenu(&menu, parent->GetClientRect().GetLeftBottom()) && commandId != wxID_NONE)
+      {
+         if(commandId == wxID_REMOVE)
+            return wxString {};
+         else if(commandId == wxID_MORE)
+            OpenInDefaultBrowser("https://plugins.audacityteam.org/");
+         else
+            return visitor.GetPluginID(commandId).GET();
+      }
+      
+      return {};
+   }
+   
+   void UpdateEffectMenuItems()
+   {
+      auto root = std::make_unique<MenuTable::MenuItem>(Identifier {}, TranslatableString {});
+
+      static auto realtimeEffectPredicate = [](const PluginDescriptor& desc)
+      {
+         return desc.IsEffectRealtime();
+      };
```

Comment:
You made the suggested simplification to remove duplication.

Amusingly:  you could even simplify this more to:

```
      static auto realtimeEffectPredicate = &PluginDescriptor::IsEffectRealtime;
```

The implicit constructor of `std::function` used when you call `MenuHelper::PopulateEffectsMenu` works even when the template parameter is of type pointer-to-memner-function.